src/features/references/editor/ReferenceDetailsCard.tsx

```diff
@@ -0,0 +1,267 @@
+import { ReactElement, useCallback, useState } from 'react';
+
+import { CloseIcon, EditIcon } from '../../../components/icons';
+import { cx } from '../../../lib/cx';
+import { Author, ReferenceItem } from '../../../types/ReferenceItem';
+import { authorsFormatter } from './grid/formatters';
+
+const getContentString = (id: string, content?: string | Author[]): string => {
+  if (typeof content == 'undefined') {
+    return '';
+  } else if (typeof content != 'string' && id === 'authors') {
+    return authorsFormatter({ value: content });
+  } else if (id === 'citationKey') {
+    return '[' + content.toString() + ']';
+  } else {
+    return content.toString();
+  }
+};
+
+const DataTextInput = ({
+  content,
+  id,
+  onChangeHandler,
+}: {
+  content: string;
+  id: string;
+  onChangeHandler: CallableFunction;
+}) => {
+  const [value, setValue] = useState(content);
+
+  return (
+    <input
+      className="w-full border bg-slate-50 px-2 py-0.5"
+      name={id}
+      title={id} // required for unit testing to find this element
```

Comment:
I don't think that's quite right either. getByTestId should be a last resort but findByRole was not working as expected without a matching title attribute (for some reason??). In order to make these more resilient and more "like a user" would use the app I'm now using findByLabelText and globally unique IDs to the inputs so "for" attributes in the labels work. (Globally unique so that even if we end up putting more than one editable reference details card on the screen at a time the labels will still be associated with the correct inputs). This also improves accessibility.